homeassistant/components/litterrobot/__init__.py

```diff
@@ -0,0 +1,50 @@
+"""The Litter-Robot integration."""
+import asyncio
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+
+from .const import DOMAIN
+from .hub import LitterRobotHub
+
+PLATFORMS = ["vacuum"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Litter-Robot component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Litter-Robot from a config entry."""
+    hub = hass.data[DOMAIN][entry.entry_id] = LitterRobotHub(hass, entry.data)
+    try:
+        await hub.login(load_robots=True)
+    except Exception as ex:
```

Comment:
Only trap the exceptions we expect.

If the credentials are invalid do not raise ConfigEntryNotReady so it doesn't try to login over and over.

Log an error and return `False` instead.  In a future PR add https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#reauthentication